cake-autorate.sh

```diff
@@ -1688,14 +1745,52 @@ then
 	exit 1
 fi
 
-# shellcheck source=config.primary.sh
-. "${config_path}"
+# validate config entries before loading
+mapfile -t user_config < <(grep -E '^[^(#| )].*=' "${config_path}" | sed -e 's/[\t ]*\#.*//g' -e 's/=.*//g')
+config_error_count=0
+for key in "${user_config[@]}"
+do
+	# Despite the fact that config_file_check is no longer required,
+	# we make an exemption just in this case as that variable in
+	# particular does not have any real impact to the operation
+	# of the script.
+	[[ "${key}" == "config_file_check" ]] && continue
+
+	# shellcheck disable=SC2076
+	if [[ ! " ${valid_config_entries[*]} " =~ " ${key} " ]]
+	then
+		((config_error_count++))
+		log_msg "ERROR" "The key: '${key}' in config file: '${config_path}' is not a valid config entry."
+	else
+		# shellcheck disable=SC2311
+		read -r user supposed <<< "$(validate_config_entry "${config_path}" "${key}")"
+		if [[ -n "${supposed}" ]]
+		then
+			error_msg="The value of '${key}' in config file: '${config_path}' is not a valid ${supposed} value."
+
+			case "${user}" in
+				*float)          error_msg="${error_msg} Also, floats are not supported." ;;
```

Comment:
But we use floats like in (dl/ul)_delay_thr_ms?

Reply:
They aren't?

Reply:
Oh, I see..